pkg/ingester/stream.go

```diff
@@ -113,8 +117,9 @@ type entryWithError struct {
 	e     error
 }
 
-func newStream(cfg *Config, tenant string, fp model.Fingerprint, labels labels.Labels, unorderedWrites bool, metrics *ingesterMetrics) *stream {
+func newStream(cfg *Config, limits *validation.Overrides, tenant string, fp model.Fingerprint, labels labels.Labels, unorderedWrites bool, metrics *ingesterMetrics) *stream {
 	return &stream{
+		limiter:         limiter.NewRateLimiter(newLocalStreamRateStrategy(limits), 10*time.Second),
```

Comment:
I looked at the the Cortex rate limiter utility and realized it has a tenancy concept built into it. This has some nice benefits (automatically rechecking dynamically loaded `overrides`) as well as some costs (mutex). I suspect this is low cost enough that we don't have to worry about it (we'll have a map with one entry, gated by a `RWMutex`). However, if we end up seeing profile issues here in the future, we may have to roll our own or contribute upstream.

Reply:
Easy enough to roll our own in the near future, just makes use of an actual rate limiter library under the hood. I can file an issue for this.